hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -566,6 +568,21 @@ private boolean isContainerFull(Container container) {
     }
   }
 
+  private boolean isVolumeFull(Container container) {
+    boolean isOpen = Optional.ofNullable(container)
+        .map(cont -> cont.getContainerState() == ContainerDataProto.State.OPEN)
+        .orElse(Boolean.FALSE);
+    if (isOpen) {
+      HddsVolume volume = container.getContainerData().getVolume();
+      long volumeCapacity = volume.getCapacity();
+      long volumeFreeSpaceToSpare =
+          VolumeUsage.getMinVolumeFreeSpace(conf, volumeCapacity);
+      long volumeAvailable = volume.getAvailable();
+      return (volumeAvailable <= volumeFreeSpaceToSpare);
```

Comment:
@sadanand48 ,  "- vol.getCommittedBytes()" is missing here.

Reply:
Done.

Reply:
Thanks @sadanand48 . The last patch LGTM, +1.